### ISSUE TITLE
TPU support

### ISSUE THREAD

**Issue by Ontopic** [replicate/cog#545]
Awesome initiative, props to the maintainers!

Was wondering if TPU support is anything on your radar or if contributions in that area are welcome?

**Comment by bfirsh** [comment]
Yes! Related: #376

**Comment by Ontopic** [comment]
Not sure whether I should reply on the open issue, or here. The dream of cog with TPU-support is large though. All the features you advertise, are clearly needed on the TPU. Perhaps you could even count on some assistance from Google to assist with proper integration?

Awesome project nonetheless, hoping it will bring the TPU into the fold!

**Comment by Ontopic** [comment]
I could post my findings into integrating TPU-access with Docker and recompiling libtpu for different setups? That would perhaps help someone on the Cog team to speed things along. But without someone knowledgable from your end I'm not sure I could achieve much.

**Comment by bfirsh** [comment]
Yes, that would be very helpful, thank you! Even if it's just links to information.

**Comment by devxpy** [comment]
I was trying this out, and the breaker I ran into was that cog can't run docker with `--privileged --network=host` ([As is required by TPUs](https://cloud.google.com/tpu/docs/run-in-container))

Would be nice to allow custom `docker run` params when running `cog build`!
